src/main/java/teammates/logic/core/FeedbackSessionsLogic.java

```diff
@@ -2191,6 +2101,240 @@ private FeedbackSessionResultsBundle getFeedbackSessionResultsForUserWithParams(
+     * @param relevantResponses   Mapping of response ids to their associated FeedbackResponseAttributes.
+     * @param student
+     * @param studentsEmailInTeam
+     * @param allResponseComments
+     */
+    private void mapResponsesToComments(String courseId, String userEmail, UserType.Role role,
+            CourseRoster roster, Map<String, FeedbackQuestionAttributes> relevantQuestions,
+            Map<String, List<FeedbackResponseCommentAttributes>> responseComments,
+            Map<String, FeedbackResponseAttributes> relevantResponses, StudentAttributes student,
+            Set<String> studentsEmailInTeam, List<FeedbackResponseCommentAttributes> allResponseComments) {
+        
+        for (FeedbackResponseCommentAttributes frc : allResponseComments) {
+            FeedbackResponseAttributes relatedResponse = relevantResponses.get(frc.feedbackResponseId);
+            FeedbackQuestionAttributes relatedQuestion = relevantQuestions.get(frc.feedbackQuestionId);
+            boolean isVisibleResponseComment = frcLogic.isResponseCommentVisibleForUser(userEmail, courseId,
+                    role, student, studentsEmailInTeam, relatedResponse, relatedQuestion, frc);
+            if (isVisibleResponseComment) {
+                if (!frcLogic.isNameVisibleTo(frc, relatedResponse, userEmail, roster)) {
+                    frc.giverEmail = Const.DISPLAYED_NAME_FOR_ANONYMOUS_COMMENT_PARTICIPANT;
+                }
+                
+                if (responseComments.get(frc.feedbackResponseId) == null) {
+                    responseComments.put(frc.feedbackResponseId,
+                            new ArrayList<FeedbackResponseCommentAttributes>());
+                }
+                responseComments.get(frc.feedbackResponseId).add(frc);
+            }
+        }
+        
+        for (List<FeedbackResponseCommentAttributes> responseCommentList : responseComments
+                .values()) {
+            Collections.sort(responseCommentList,
+                    new ResponseCommentCreationDateComparator());
+        }
+    }
+    
+    /**
+     * Adds a list of additional details for the given response to a series of tables.
+     * These include the name, last name, and team name of the giver,
+     * and whether the giver or receiver name is visible.
+     * 
+     * @param userEmail
+     * @param role
+     * @param roster
+     * @param emailNameTable
+     * @param emailLastNameTable
+     * @param emailTeamNameTable
+     * @param visibilityTable
+     * @param question
+     * @param response
+     */
+    protected void addResponseAdditionalDetailsToTable(String userEmail, UserType.Role role,
+            CourseRoster roster, Map<String, String> emailNameTable, Map<String, String> emailLastNameTable,
+            Map<String, String> emailTeamNameTable, Map<String, boolean[]> visibilityTable,
+            FeedbackQuestionAttributes question, FeedbackResponseAttributes response) {
```

Comment:
This method effectively has a scope of `private`, however due to the `Map<String,boolean[]>` parameter confusing PMD and flagging this as an unused private method, this was changed to `protected`. This follows the example of `addVisibilityToTable()` in this class.